src/shogun/classifier/svm/SVMLight.cpp

```diff
@@ -1595,7 +1595,7 @@ void SVMLight::update_linear_component_mkl(
 		for (int32_t n=0; n<num_kernels; n++)
 		{
 			w1[n]=1.0 ;
-			kernel->set_subkernel_weights(SGVector<float64_t>(w1, num_weights));
+			kernel->set_subkernel_weights(SGVector<float64_t>(w1, num_weights,false));
```

Comment:
this would simply result in a memory leak, as `w1` would not be released later. you should fix the root of this, i.e. how `w1` is actually created

Reply:
There is a SG_FREE(w1). I think [here](https://github.com/shogun-toolbox/shogun/issues/4871) you've mentioned that **false** would be fine because mem management is not handed over. 
I'm a bit confused

Reply:
mmm i think then you misunderstood my description of the issue....
what it was saying there that if you encounter a case where the `SGVector<float64_t>(memory, size, false)` constructor is being called that case should be ignored as that is fine...

Reply:
I've made a workaround without using w1, is it valid ? 
```
	SGVector<float64_t> out(num_weights);
		
		
		for index in num_kernels:	
			memset(out,0,num_kernels);
			out[index] = 1.0;
			kernel->set_subkernel_weights(out); 
                        / * other codes */
```

Reply:
so i'd go around actually fixing the creation of w1

Reply:
i'm not so sure what this supposed to be? a metacode? for which section? plz push code that's way more easier to review than comments

Reply:
Okay, just a second